### PR TITLE
clear user upon failed auth upon User.getCurrent

### LINKED ISSUE #212: BUG:  User#getCurrent does not handle error
Reproduce steps:
1.) Login with a user
2.) Go directly to DB and remove access token for that user
3.) Go to secure page and print isAuthenticated

Expected: isAuthenticated should be false
Actual: isAuthenticated is true

The issue can be resolved fairly easily.  The currentUserData is null because authentication fails from server side.  If currentUserData is null, we should clear the currentUserId so that isAuthenticated becomes false.

See generated code:
```
        "getCurrent": {
           url: urlBase + "/Users" + "/:id",
           method: "GET",
           params: {
             id: function() {
              var id = LoopBackAuth.currentUserId;
              if (id == null) id = '__anonymous__';
              return id;
            },
          },
          interceptor: {
            response: function(response) {
              LoopBackAuth.currentUserData = response.data;
              return response.resource;
            }
          },
          __isGetCurrentUser__ : true
        }
      }
    );
```

### PR DESCRIPTION
**Pull request by davidcheung**
fixes #212

### CODE REVIEW

**Comment by davidcheung** [comment]
@slnode test please

**Comment by davidcheung** [comment]
@slnode test please

**Comment by davidcheung** [comment]
@slnode test please

**Comment by davidcheung** [comment]
@bajtos PTAL

File: test.e2e/spec/services.spec.js [hunk 1]
```diff
@@ -655,6 +655,26 @@ define(['angular', 'given', 'util'], function(angular, given, util) {
...
+          .then(function(loggedInUser) {
+            return Customer.delete({ id: loggedInUser.userId }).$promise;
+          })
+          .then(function(deleteResponse) {
+            return Customer.getCurrent().$promise;
+          })
+          .then(function(res) {
+            throw new Error('User.getCurrent() was supposed to fail.');
+          }
+          , function(res) {
```
Comment (bajtos) [inline test.e2e/spec/services.spec.js]: Nitpick: I find this formatting weird. We use one-arg-per-line rule in the rest of our codebase, see https://github.com/strongloop/loopback-contributor-docs/blob/45e4dfc18718bf21077fd43053e08464b111acf3/style-guide.md#one-arg-per-line.

Comment (davidcheung) [inline test.e2e/spec/services.spec.js]: fixed, was thinking eslint would catch me 😢

**Pull request merged**